utils/src/main/java/com/cloud/utils/nio/NioConnection.java

```diff
@@ -127,17 +129,31 @@ public void start() throws NioConnectionException {
         _isStartup = true;
 
         if (_executor.isShutdown()) {
-            _executor = new ThreadPoolExecutor(_workers, 5 * _workers, 1, TimeUnit.DAYS, new LinkedBlockingQueue<>(), new NamedThreadFactory(_name + "-Handler"));
+            _executor = new ThreadPoolExecutor(_workers, 5 * _workers, 1, TimeUnit.DAYS,
+                    new LinkedBlockingQueue<>(5 * _workers), new NamedThreadFactory(_name + "-Handler"),
+                    new ThreadPoolExecutor.AbortPolicy());
+        }
+        if (_sslHandshakeExecutor.isShutdown()) {
+            String sslHandshakeHandlerName = _name + "-SSLHandshakeHandler";
+            if (factoryMaxNewConnectionsCount > 0) {
+                _sslHandshakeExecutor = new ThreadPoolExecutor(0, this.factoryMaxNewConnectionsCount, 30,
+                        TimeUnit.MINUTES, new SynchronousQueue<>(), new NamedThreadFactory(sslHandshakeHandlerName),
+                        new ThreadPoolExecutor.AbortPolicy());
+            } else {
+                _sslHandshakeExecutor = Executors.newCachedThreadPool(new NamedThreadFactory(sslHandshakeHandlerName));
+            }
```

Comment:
same code in the class constructor, can be refactored

Reply:
done